comp/dogstatsd/server/parse.go

```diff
@@ -50,15 +50,15 @@ type parser struct {
 	readTimestamps bool
 }
 
-func newParser(float64List *float64ListPool) *parser {
-	stringInternerCacheSize := config.Datadog.GetInt("dogstatsd_string_interner_size")
-	readTimestamps := config.Datadog.GetBool("dogstatsd_no_aggregation_pipeline")
+func newParser(cfg config.ConfigReader, float64List *float64ListPool) *parser {
```

Comment:
Nit and unrelated to your PR but I'm wondering if `ConfigReader` is the best name here, I would have expected something like ConfigProvider or similar.

Reply:
I agree this is a little weird but there is a reason.
`ConfigReader` is the interface for all "read" operations on the config. 
A config instance is made up of `ConfigReader` + `ConfigWriter`. 

From the component interface, [we only expose the ConfigReader](https://github.com/DataDog/datadog-agent/blob/brian/fix-auto-multiline-regex-pattern/comp/core/config/component.go#L31). As the "writer" part happens privately. 

With this in mind, the reason I pass around a `ConfigReader` [is because of the serverless workaround in](https://github.com/DataDog/datadog-agent/blob/brian/fix-auto-multiline-regex-pattern/comp/dogstatsd/server/server.go#L185) `server.go`. `ConfigReader` is the only common interface between `config.Datadog` and the `Component`. 

Once this workaround is removed we can optionally replace all references to `ConfigReader` with `Component`. (but functionally it's all the same).